src/main/java/com/patreon/PatreonOAuth.java

```diff
@@ -0,0 +1,68 @@
+package com.patreon;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.jsoup.Jsoup;
+
+import java.io.IOException;
+
+public class PatreonOAuth {
+    private final String clientID;
+    private final String clientSecret;
+    private static final Gson gson = new GsonBuilder().serializeNulls().enableComplexMapKeySerialization().create();
+
+
+    public PatreonOAuth(String clientID, String clientSecret) {
+        this.clientID = clientID;
+        this.clientSecret = clientSecret;
+    }
+
+    public String getDefaultOauthUrl(String redirectUri) {
+        return "https://www.patreon.com/oauth2/authorize?response_type=code&client_id=" + clientID + "&redirect_uri=" + redirectUri;
```

Comment:
can we use <https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/client/utils/URIBuilder.html#addParameter(java.lang.String,%20java.lang.String)> to prevent arbitrary `redirectUri`?

Reply:
same for `PatreonAPI.java` urls?

Reply:
it might be nice to use `URIBuilder` to construct the oauth flow urls as well, and then pass the result into `Jsoup.connect`